lighthouse-core/audits/nosniff.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview Audit which reports all network resources that failed to include
+ * a `X-Content-Type-Options: nosniff` header.
+ * https://docs.google.com/document/d/1KQrbenS4s7qZsptCChZCqA45v0Q7Mv9a5tVZACLFCWc/edit?usp=sharing
+ */
+
+const Audit = require('./audit.js');
+const i18n = require('../lib/i18n/i18n.js');
+const NetworkRecords = require('../computed/network-records.js');
+const URL = require('./../lib/url-shim.js');
+
+const UIStrings = {
+  /** Title of a diagnostic LH audit that provides details on network resources that do not set a `X-Content-Type-Options: nosniff` response header. */
+  title: 'Sets `X-Content-Type-Options: nosniff` and `Content-Type` headers ' +
+    'for every network resource',
+  /** Title of a Lighthouse audit that tells the user that their site contains a vaild touch icon. This descriptive title is shown when the page does not contain a valid iOS touch icon. "apple-touch-icon" is an HTML attribute value and should not be translated. */
+  failureTitle: 'Does not set a `X-Content-Type-Options: nosniff` and `Content-Type` headers ' +
+    'for every network resource',
+  /** Description of a diagnostic LH audit that shows the network resources that do not set a `X-Content-Type-Options: nosniff` response header. */
+  description: 'Network resources that do not set a `nosniff` response header may result in ' +
+    'security issues. [Learn more](https://web.dev/security-headers/#xcto)',
```

Comment:
Are there other vulnerabilities not mentioned in the link?

```suggestion
  description: 'Network resources that do not set a `nosniff` response header can sometimes ' +
    'be exploited for XSS attacks. [Learn more](https://web.dev/security-headers/#xcto)',
```